packages/docs/site/docs/11-architecture/18-host-your-own-playground.md

```diff
@@ -0,0 +1,169 @@
+# Host your own Playground
+
+You can host the Playground on your own domain instead of `playground.wordpress.net`.
+
+This is useful for having full control over its content and behavior, as well as removing dependency on a third-party remote endpoint. It can provide a customized user experience, for example: preinstalled plugins, default theme, site settings, and demo content.
+
+Also, if you expect to see heavy traffic, it's good etiquette to provide your own server/CDN resources to serve the static file assets, rather than relying on the `wordpress.net` server.
+
+#### Usage
+
+A self-hosted Playground can be embedded as an iframe.
+
+```html
+<iframe src="https://my-playground.com"></iframe>
+```
+
+Or dynamically loaded by passing the remote URL to the Playground Client.
+
+```ts
+import { startPlaygroundWeb } from '@wp-playground/client';
+
+const client = await startPlaygroundWeb({
+	iframe: document.getElementById('wp'),
+	remoteUrl: `https://my-playground.com/remote.html`,
+});
+```
+
+## Build static assets
+
+Create a shallow clone of the Playground repository.
+
+```sh
+git clone -b trunk --single-branch --depth 1 git@github.com:WordPress/wordpress-playground.git
+```
+
+Install dependencies, and build the website.
+
+```sh
+npm install
+npm run build:website
+```
```

Comment:
is there a way to keep these internal details out of the docs?
maybe we need a Github Actions button that runs these and then produces an artifact with the required files, that way someone could clone the repo, click the button, and then get their files without having to clone or install or do anything else.

Reply:
We probably can't completely avoid mentioning these internal details altogether, since they're part of the build process used by the project and any developer working on it.

These two commands specifically are the most straight-forward way to install project dependencies and build the website. The task `build:website` is provided as part of a de-facto public API of all NPM scripts defined in `package.json`, including `npm run dev`, `npm run build`, `npm run build:docs`, etc. It seems inevitable that if any of these tasks are renamed, they would have to be renamed everywhere in the docs.

But I had the same thought about other parts of the docs, including this new page, that it's easy for the information to become out of date, out of sync with the codebase or project/package build scripts. For example, the generated artifact of the `build:website` task is currently in a folder named `wasm-wordpress-net`. This name is likely to change in the future, since it's an implementation detail that users don't need to know about - probably the name is a historical vestige from an earlier phase in the development of the Playground, when it was hosted at `wasm.wordpress.net`. Anyway, if the internal build task and the destination folder is renamed, the documentation needs to be updated anywhere that name appears.

A possible way to keep the docs synced automatically with the codebase and build scripts, might be to use some kind of variable or token in the Markdown source.

Reply:
We're already exposing that as GitHub artifacts from the deploy website workflow (that will now run each day):

https://github.com/WordPress/wordpress-playground/actions/workflows/build-website.yml

It's not an ideal link to point someone to, but it's worth noting it's there.

Reply:
I was inspired on this from the https://github.com/zmkfirmware/zmk repo, where the model is to fork and hit the Actions button. You don't need to know any names or have any software installed on your system. In fact, actually I think the way they do it is on push they generate the artifacts automatically.

This would be something like "Download the artifact from [the latest build website action](https://github.com/WordPress/wordpress-playground/actions/workflows/build-website.yml?query=is%3Asuccess) or fork the repo, push your changes, and download the artifact from there…"

It was more the things like installing the dependencies, knowing about `wasm-wordpress-net` and the like that make me skeptical here.

Maybe good enough for now but it's not a great workflow if all someone wants to do is copy the existing static files.

Can we create a link to the latest successful build artifact for the workflow? 🤔

Reply:
There may be different types of target audience for this documentation page.

Those who want to self-host the Playground as is, can download the prebuilt artifact from the GitHub action/workflow, then unzip and upload it to their server. Ideally, it would be nice to include that exact download link right there in the documentation for the latest successful build.

Other people might want to customize the Playground by editing the `website` package for how the embedded iframe looks and behaves. For them, I think it's useful to include information about how to locally build the website themselves.

Currently this information is implicit in the build scripts, like the fact that the complete static site assets are in the folder `wasm-wordpress-net`. That's not written anywhere, and it took me a while to figure this out when I was learning how to build and host the Playground. But it could be that this audience is very small, or even just one person (me).

Reply:
> Download the artifact from [the latest build website action](https://github.com/WordPress/wordpress-playground/actions/workflows/build-website.yml?query=is%3Asuccess) or **fork the repo, push your changes, and download the artifact** from there

I tried this latter method by forking the Playground repo, then going to: Actions -> Deploy to playground.wordpress.net -> Run workflow.

For some reason it just skips the build process.

![image](https://github.com/WordPress/wordpress-playground/assets/5352434/1f2c483d-6262-4fda-b7a8-c3aa71a6a8ab)

I'm sure this is the correct GitHub action (`build-website.yml`) but its job title makes it sounds like it could accidentally try to deploy to `playground.wordpress.net` for real.

Oh I see, this is probably why it's skipped:

```
# Only run this workflow from the trunk branch and when it's triggered by another workflow OR dmsnell OR adamziel
if: >
    github.ref == 'refs/heads/trunk' && (
        github.event_name == 'workflow_run' ||
        github.actor == 'adamziel' ||
        github.actor == 'dmsnell'
    )
```

---

Well, for now, I think this documentation page could show the possible methods of building static file assets:

- Download the artifact from [the latest build website action](https://github.com/WordPress/wordpress-playground/actions/workflows/build-website.yml?query=is%3Asuccess)
- Fork the repo and
  - Build locally: `npm install` and `npm run build:website`
  - Build using GitHub action: Actions -> Deploy to playground.wordpress.net -> Run workflow